java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -0,0 +1,250 @@
+                                        + "    @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ObjectMeta.class),\n"
+                                        + "    @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ObjectReference.class),\n"
+                                        + "    @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.LabelSelector.class),\n"
+                                        + "    @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.Container.class),\n"
+                                        + "    @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.EnvVar.class),\n"
+                                        + "    @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.ContainerPort.class),\n"
+                                        + "    @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.Volume.class),\n"
+                                        + "    @io.sundr.builder.annotations.BuildableReference(io.fabric8.kubernetes.api.model.VolumeMount.class)\n"
+                                        + "}")));
+
+        clz.addImplementedType("io.fabric8.kubernetes.api.model.KubernetesResource");
+
+        if (this.options.isPreserveUnknownFields()) {
+            if (!clz.getFieldByName(ADDITIONAL_PROPERTIES).isPresent()) {
+                ClassOrInterfaceType mapType =
+                        new ClassOrInterfaceType()
+                                .setName(JAVA_UTIL_MAP)
+                                .setTypeArguments(
+                                        new ClassOrInterfaceType().setName("String"),
+                                        new ClassOrInterfaceType().setName("Object"));
+                FieldDeclaration objField =
+                        clz.addField(mapType, ADDITIONAL_PROPERTIES, Modifier.Keyword.PRIVATE);
+                objField.setVariables(
+                        new NodeList<>(
+                                new VariableDeclarator()
+                                        .setName(ADDITIONAL_PROPERTIES)
+                                        .setType(mapType)
+                                        .setInitializer(
+                                                "new java.util.HashMap<String, Object>()")));
+
+                objField.addAnnotation("com.fasterxml.jackson.annotation.JsonIgnore");
+
+                objField.createGetter()
+                        .addAnnotation("com.fasterxml.jackson.annotation.JsonAnyGetter");
+                objField.createSetter()
+                        .addAnnotation("com.fasterxml.jackson.annotation.JsonAnySetter");
+            } else {
+                // Warning ???
+            }
+        }
+
+        List<String> buffer = new ArrayList<>(this.fields.size() + 1);
+
+        // CU to expand inner Enums
```

Comment:
s/Enums/types/ ?

Reply:
For now the only thing we are expanding here are Enums